### DISCUSSION TITLE
Simplify base interface member override

### DISCUSSION THREAD

**Discussion by Rekkonnect** [dotnet/csharplang#6395] [Language Ideas]
I'm toying with the new static abstracts feature, and I'm loving how I can statically type all my implementations without bothering about singleton instances and potential overhead by that. However, my code for writing those implementations looks junky even for the simplest of things, like defining names in their appropriate places.

I'm currently implementing some basic information about how to retrieve data from an API using this design pattern through static abstracts, and my current code looks like this:

```csharp
    private interface IEndpointController<T>
        where T : IEndpointController<T>
    {
        public static abstract string Name { get; }
    }

    private interface IRoutesController : IEndpointController<IRoutesController>
    {
        static string IEndpointController<IRoutesController>.Name => "getRoutes";
    }
    private interface IStopsController : IEndpointController<IStopsController>
    {
        static string IEndpointController<IStopsController>.Name => "getStops";
    }
    private interface IRouteStopsController : IEndpointController<IRouteStopsController>
    {
        static string IEndpointController<IRouteStopsController>.Name => "getRouteStops";
    }
    private interface ILinesController : IEndpointController<ILinesController>
    {
        static string IEndpointController<ILinesController>.Name => "getLines";
    }
    private interface IMasterlinesDetailsController : IEndpointController<IMasterlinesDetailsController>
    {
        static string IEndpointController<IMasterlinesDetailsController>.Name => "getMasterlinesDetails";
    }
```

*I'm planning to extend these to defining data parsing, and concretely typing the returning type's name, which will further bloat this.*

I feel that specifying the base interface quickly becomes a pain in this design pattern, and I'd like any of the two options:

```csharp
    private interface IRoutesController : IEndpointController<IRoutesController>
    {
        static string base.Name => "getRoutes"; // Option A
        static override string Name => "getRoutes"; // Option B
    }
```

Option B could also be extended to allow specifying the accessibility modifier (`public`), or implying it through the base. It would obviously be an error to declare a different accessibility modifier than that of the base declaration.

**Comment by HaloFour** [comment]
Note that this isn't specific to `static abstract`, this is how default implementations for a member on a base interface has always worked:

```cs
public interface IFoo {
    void M();
}

public interface IBar : IFoo {
    void IFoo.M() { }
}
```

See: https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/default-interface-methods.md#explicit-implementation-in-interfaces

**Reply by Rekkonnect** [reply]
Right, this is the behavior across all DIMs.

The proposal could either only apply to static abstracts, or DIMs in general, which generally need some improvements.
